exits immediately on macos

Hi! Great project, thank you. Tried to use it but faced an issue. No logs provided, nothing. Just closes and that's it.

Used command:

```
  13:29:02 ➜  exec node "/Users/colch/.npm/_npx/9dae106e3b8384cc/node_modules/interactive-mcp/dist/commands/input/ui.js" "c0769e1cec08863a"; exit 0
```

in cline, I see that llm uses this mcp:

```
{
  "projectName": "homelab-ansibles",
  "message": "Did you see the interactive MCP popup? (simple yes/no test)"
}
```

then terminal window opens and immediately closes. In Cline I see:

```
User replied with empty input.
```

How to approach to this? Having any logs in the terminal would be really helpful... 

If this is important, I use 'fish shell'.

Cline's 'c custom instruction used:

```
# Interactive-MCP strategy

You are an AI assistant integrated with an MCP-compatible environment using the `interactive-mcp` protocol. **Do not proceed with any action if the user's intent is ambiguous or unclear.** Instead, use the appropriate `interactive-mcp` tools to ask clarifying questions before taking further steps.

Use these protocols as follows:

- If the task is unclear or could be interpreted in more than one way, use `request_user_input` to prompt the user via terminal.
- For multi-step clarification, use the sequence `start_intensive_chat`, `ask_intensive_chat`, and `stop_intensive_chat`.
- After completing a task, notify the user via `message_complete_notification`.

Your goal is to reduce errors, avoid unnecessary API calls, and maintain a back-and-forth dialogue with the user instead of guessing. **Always prioritize user confirmation over assumption.**

When you need clarification, send:  
```json
{
  "tool": "request_user_input",
  "message": "[Your question here]",
  "options": [Optional predefined options]
}

# Interaction

- Please use the interactive MCP tools
- Please provide options to interactive MCP if possible

# Reduce Unexpected Changes

- Do not make assumption.
- Ask more questions before executing, until you think the requirement is clear enough.
```

and "task" prompt:

```
do you see "interactive mcp" and custom instructions ??? ask me some question using interactive mcp, lets test it
```

Same with the latest Cursor 1.0, wasn't an issue with the previous version.

On linux (Debian 12).

Cursor > Output > MCP Logs:

```
2025-06-05 14:50:56.249 [info] tive: Connected to stdio server, fetching offerings
2025-06-05 14:50:56.262 [info] listOfferings: Found 5 tools
2025-06-05 14:50:56.263 [info] tive: Found 5 tools
2025-06-05 14:52:44.798 [info] tive: Handling CallTool action for tool 'request_user_input'
2025-06-05 14:52:44.798 [info] tive: Calling tool 'request_user_input'
2025-06-05 14:52:52.810 [info] tive: Successfully called tool 'request_user_input'
```

## Related - user-prompt-mcp

The other similar MCP - user-prompt-mcp - also does not display any dialogs and gives the error:

```
2025-06-05 14:32:59.937 [error] user-prompt: [UserPromptMCP] 2025/06/05 14:32:59 mcp_server.go:35: Request: [4] tools/call
[UserPromptMCP] 2025/06/05 14:32:59 mcp_server.go:96: User prompt request: prompt="OK?", title=""

2025-06-05 14:33:00.082 [error] user-prompt: [UserPromptMCP] 2025/06/05 14:33:00 mcp_server.go:105: Error getting user input: user cancelled input
[UserPromptMCP] 2025/06/05 14:33:00 mcp_server.go:43: Error: [4] tools/call: request error: failed to get user input: user cancelled input
```

So probably it's Cursor 1.0 issue.

--- a/src/commands/input/index.ts
+++ b/src/commands/input/index.ts
@@ -94,7 +94,7 @@ export async function getCmdWindowInput(
 
           // Construct the command string directly for the shell. Quotes handle paths with spaces.
           // Pass only the sessionId
-          const nodeCommand = `exec node "${escapedScriptPath}" "${escapedSessionId}"; exit 0`;
+          const nodeCommand = `exec node "${escapedScriptPath}" "${escapedSessionId}" "${tempDir}"; exit 0`;
 
           // Escape the node command for osascript's AppleScript string:
           const escapedNodeCommand = nodeCommand

--- a/src/commands/input/ui.tsx
+++ b/src/commands/input/ui.tsx
@@ -37,7 +37,11 @@ const readOptionsFromFile = async (): Promise<CmdOptions> => {
     throw new Error('No sessionId provided'); // Throw error to prevent proceeding
   }
 
-  const tempDir = os.tmpdir();
+  let tempDir = args[1];
+  if (!tempDir) {
+      tempDir = os.tmpdir();
+  }
+
   const optionsFilePath = path.join(
     tempDir,
     `cmd-ui-options-${sessionId}.json`,

